### PR TITLE
Refactor logic to get plugins path for android studio in mac

### LINKED ISSUE #26928: flutter doctor fails to detect flutter/dart plugins in second Android Studio 3.4 preview instance
Hi,

I have two local instances of Android Studio: 3.3 and 3.4 Preview. Both have the latest available Flutter and Dart plugins installed.

When I run `flutter doctor`, it consistently fails to detect the plugins in the second instance.

```
flutter doctor -v
[✓] Flutter (Channel beta, v1.0.0, on Mac OS X 10.14.2 18C54, locale en-US)
    • Flutter version 1.0.0 at /Users/julien.cayzac/github/flutter/flutter
    • Framework revision 5391447fae (8 weeks ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/julien.cayzac/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/julien.cayzac/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.3

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 3.4 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b22)

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 31.3.3
    • Dart plugin version 182.5124
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] IntelliJ IDEA Community Edition (version 2018.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 31.3.4
    • Dart plugin version 183.4886.3

[!] Connected device
    ! No devices available

! Doctor found issues in 1 category.
```

### PR DESCRIPTION
**Pull request by kangwang1988**
Refactor logic to get plugins path for android studio in Mac.
This will fix https://github.com/flutter/flutter/issues/26928
It's better to use 'idea.paths.selector' from Info.plist instead of combine major/minor version.
See: https://github.com/JetBrains/intellij-community/blob/master/platform/util/src/com/intellij/openapi/application/PathManager.java

### CODE REVIEW

**Comment by kangwang1988** [comment]
@joaoatgoogle 
Could you have a look at this PR?

**Comment by joaoatgoogle** [comment]
I'm not an appropriate reviewer for this change

**Review by chinmaygarde — APPROVED** [review]

File: packages/flutter_tools/lib/src/android/android_studio.dart [hunk 1]
```diff
@@ -47,7 +49,13 @@ class AndroidStudio implements Comparable<AndroidStudio> {
     Version version;
     if (versionString != null)
       version = Version.parse(versionString);
-    return AndroidStudio(studioPath, version: version);
+
+    final String plistValue = iosWorkflow.getPlistValueFromFile(
+      plistFile,
+      null,
+    );
+    final String pathsSelectorValue = _pathsSelectorMatcher.stringMatch(plistValue).split('=').last.trim().replaceAll('"', '');
```
Comment (xster) [inline packages/flutter_tools/lib/src/android/android_studio.dart]: @chinmaygarde, somewhat tangential but you added a note to not use PlistBuddy in #1969 but we still have a wrapper in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/ios/mac.dart#L38. Should we remove it?

File: packages/flutter_tools/lib/src/android/android_studio.dart [hunk 2]
```diff
@@ -83,6 +91,7 @@ class AndroidStudio implements Comparable<AndroidStudio> {
   final String studioAppName;
   final Version version;
   final String configured;
+  final String pathsSelectorPath;
```
Comment (xster) [inline packages/flutter_tools/lib/src/android/android_studio.dart]: If it isn't named in a platform specific way, might as well extract lines 109-119 out into the constructors. And add dartdocs for the constructors for targeted platforms.

**Comment by xster** [comment]
Could you add a test?

**Comment by kangwang1988** [comment]
> Could you add a test?

Done in: https://github.com/flutter/flutter/pull/27944

**Pull request merged**